packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/Functions.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.kotlin.mongodb
+
+import io.realm.kotlin.mongodb.exceptions.AppException
+import io.realm.kotlin.mongodb.internal.BsonEncoder
+import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.serializer
+import org.mongodb.kbson.BsonArray
+import org.mongodb.kbson.BsonDocument
+import org.mongodb.kbson.BsonValue
+
+public interface Functions {
+    public val app: App
+    public val user: User
+
+    /**
+     * Invokes an App Services Application function.
+     *
+     * Due to the serialization engine does not support third-party libraries yet, there are some
+     * limitations in what types can be used as arguments and return types:
+     *
+     * - Primitives, Bson, lists, and maps are valid argument types.
+     * - Results can only be deserialized to primitives or Bson types.
+     *
+     * The Bson implementations for arrays or maps are [BsonArray] and [BsonDocument], and they can be
+     * used as valid return types.
+     *
+     * @param name Name of the function to call.
+     * @param args Arguments to the function.
+     * @param deserializationStrategy Deserialization strategy for decoding the results.
+     * @param T The type for the functions response.
+     * @return Result of the function.
+     *
+     * @throws AppException if the request failed in some way.
+     */
+    public suspend fun <T : Any?> invoke(
```

Comment:
Do we want to expose calling functions like this? 
At least I find the syntax `app.currentUser.functions.call(...)` more readable than `app.currentUser.functions(...)`